store/mongo/datastore_mongo.go

```diff
@@ -303,12 +300,12 @@ func (db *DataStoreMongo) AddToken(ctx context.Context, t *jwt.Token) error {
 	collTokens := database.Collection(DbTokensColl)
 
 	filter := bson.M{"_id": t.Claims.ID}
-	update := bson.M{"$set": t}
+	update := bson.M{"$set": ctxstore.WithTenantID(ctx, t)}
 	updateOpts := mopts.Update()
 	updateOpts.SetUpsert(true)
 
 	if _, err := collTokens.UpdateOne(
-		ctx, filter, update, updateOpts,
+		ctx, ctxstore.WithTenantID(ctx, filter), update, updateOpts,
```

Comment:
Token claims should not have another tenant_id field. (It's already present in `mender.tenant` field).
```suggestion
		ctx, filter, update, updateOpts,
```

Reply:
`mender.tenant` wouldn't work because of struct `inline` and `omitempty` bson tags. Currently `tenant` field is used to filter tokens.